### DISCUSSION TITLE
Interrupting solver on Windows PowerShell

### DISCUSSION THREAD

**Discussion by Mapiarz** [google/or-tools#3144] [CP-SAT questions]
Hi! I have a python program with CP-SAT solver. On linux, I can press ctrl+c once to interrupt the solver and then continue exceution of the remainder of the program (e.g. print the latest solution).

On windows, pressing ctrl+c immediately terminates the program. Is there any way I can mimic the linux behaviour on windows?

Thanks

**Comment by lperron** [comment]
send a PR to implement the complex sigint processing on windows. The code is in ortools/util/sigint.cc :-)

**Comment by Mapiarz** [comment]
What about working this around at application level? I tried running the solver in a background thread (using `asyncio` and `ThreadPoolExecutor`) and that works, but when I call `solver.StopSearch`, nothing seems to happen and the solver continues to search for solutions.

Any ideas?

**Comment by lperron** [comment]
No idea, the sample StopAfterNSolutionsSampleSat.cs works.
So the stopping mechanism works.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00
